### ISSUE TITLE
Split up the e2e testing GitHub action to speed up receiving test results

### ISSUE THREAD

**Issue by prestoncarman** [Submitty/Submitty#8611]
**What problem are you trying to solve with Submitty**
The Submitty tests running on GitHub take over an hour to run. Primarily, due to one test that incorporates many different types of tests into a single threaded process.

**Describe the way you'd like to solve this problem**
Instead of using one action on distinct tests, the action could be split up to run in parallel and improve overall test suite running time. The total test time will not change, but results will be produced quicker. The e2e test could be split into these testing areas:

* Selenium tests
* Cypress test part 1
* Cypress test part 2
* Integration tests

**Additional context**
Here is a slack comment from @williamjallen:

> For the ~50-minute run I linked above, ~7 minutes were spent running Selenium tests, ~15 minutes were spent running Cypress tests, and ~10 minutes were spent running integration tests.  Those three categories combined to make up about 70% of the total running time.

And a follow up from @shailpatels:

> ... I can definitely see making an e2e job for selenium and just for cypress to get them in parallel. We could also split cypress into a subset of tests between 2 jobs as well

You can find the full thread on slack <https://submitty.slack.com/archives/C9A0XMXEX/p1666843464571729>. These comments come towards the end of the thread.

**Comment by bmcutler** [comment]
possible duplicate of #6390
